In [ ]:
from google.colab import files
uploaded = files.upload()

# **Prepare data**

In [ ]:
import pandas as pd
import numpy as np

df = pd.read_csv('new_fused_dataset.csv')
df.head()

suicides/100k pop      Status  Life expectancy  Adult Mortality  \
0               8.27  developing             76.2               91   
1               5.38  developing             76.2               91   
2               5.01  developing             76.2               91   
3               3.94  developing             76.2               91   
4               3.58  developing             76.2               91   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   BMI  \
0              1     5.28               41.822757           99       10  54.3   
1              1     5.28               41.822757           99       10  54.3   
2              1     5.28               41.822757           99       10  54.3   
3              1     5.28               41.822757           99       10  54.3   
4              1     5.28               41.822757           99       10  54.3   

   ...  Polio  Total expenditure  Diphtheria  HIV/AIDS         GDP  \
0  ...     99               5.34          99       0.1  494.358832   
1  ...     99               5.34          99       0.1  494.358832   
2  ...     99               5.34          99       0.1  494.358832   
3  ...     99               5.34          99       0.1  494.358832   
4  ...     99               5.34          99       0.1  494.358832   

   thinness 5-9 years  Income composition of resources  Schooling  \
0                 1.5                            0.725       12.5   
1                 1.5                            0.725       12.5   
2                 1.5                            0.725       12.5   
3                 1.5                            0.725       12.5   
4                 1.5                            0.725       12.5   

   HDI for year  gdp_per_capita ($)  
0      -1.18134           -0.679695  
1      -1.18134           -0.679695  
2      -1.18134           -0.679695  
3      -1.18134           -0.679695  
4      -1.18134           -0.679695  

[5 rows x 21 columns]

In [ ]:
#check if there are missing values
df.isnull().sum()

#check the data frame info
df.info()

#calculate the descriptive statistics
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15264 entries, 0 to 15263
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   suicides/100k pop                15264 non-null  float64
 1   Status                           15264 non-null  object 
 2   Life expectancy                  15264 non-null  float64
 3   Adult Mortality                  15264 non-null  int64  
 4   infant deaths                    15264 non-null  int64  
 5   Alcohol                          15264 non-null  float64
 6   percentage expenditure           15264 non-null  float64
 7   Hepatitis B                      15264 non-null  int64  
 8   Measles                          15264 non-null  int64  
 9   BMI                              15264 non-null  float64
 10  under-five deaths                15264 non-null  int64  
 11  Polio                            15264 non-null  int64  
 12  Total expenditure 

suicides/100k pop  Life expectancy  Adult Mortality  infant deaths  \
count       15264.000000     15264.000000     15264.000000   15264.000000   
mean           12.217648        75.591431       115.023585       4.901730   
std            18.203273         5.458163        75.759019      12.373685   
min             0.000000        53.700000         1.000000       0.000000   
25%             0.910000        72.700000        66.000000       0.000000   
50%             5.500000        75.400000        98.500000       1.000000   
75%            15.642500        79.300000       162.000000       3.000000   
max           204.920000        89.000000       498.000000     111.000000   

            Alcohol  percentage expenditure   Hepatitis B       Measles  \
count  15264.000000            15264.000000  15264.000000  15264.000000   
mean       7.014623             1522.914475     87.151730    550.509434   
std        3.694886             2807.340450     20.837042   2715.973806   
min        0.010000                0.000000      2.000000      0.000000   
25%        3.767500               47.491680     91.000000      0.000000   
50%        7.215000              367.592120     93.000000      2.000000   
75%        9.970000             1346.941343     97.000000     55.000000   
max       17.870000            19479.911610     99.000000  42724.000000   

                BMI  under-five deaths         Polio  Total expenditure  \
count  15264.000000       15264.000000  15264.000000       15264.000000   
mean      47.424686           5.955189     91.150157           6.537787   
std       17.329782          15.300963     16.900051           2.556611   
min        2.200000           0.000000      7.000000           1.100000   
25%       43.000000           0.000000     92.000000           5.125000   
50%       54.200000           1.000000     96.000000           6.525000   
75%       58.500000           3.000000     98.000000           8.250000   
max       69.600000         127.000000     99.000000          17.600000   

         Diphtheria      HIV/AIDS            GDP  thinness 5-9 years  \
count  15264.000000  15264.000000   15264.000000        15264.000000   
mean      91.650157      0.397406   12375.351447            2.619811   
std       15.641716      2.335284   17999.797864            2.501877   
min        5.000000      0.100000       1.681350            0.200000   
25%       92.000000      0.100000    1766.947595            1.000000   
50%       96.000000      0.100000    4354.998196            2.000000   
75%       98.000000      0.100000   14215.013397            3.300000   
max       99.000000     29.700000  119172.741800           19.100000   

       Income composition of resources     Schooling  HDI for year  \
count                     15264.000000  15264.000000  1.526400e+04   
mean                          0.750106     14.118160 -2.670719e-02   
std                           0.165436      2.471336  9.407779e-01   
min                           0.000000      0.000000 -3.983621e+00   
25%                           0.698000     12.600000  1.830000e-15   
50%                           0.775000     14.000000  1.830000e-15   
75%                           0.853000     15.800000  1.830000e-15   
max                           0.948000     20.700000  2.478110e+00   

       gdp_per_capita ($)  
count        15264.000000  
mean             0.168259  
std              1.137096  
min             -0.887560  
25%             -0.659992  
50%             -0.292761  
75%              0.702407  
max              5.587749

# **One-hot encoding for 'status' column**


*   Status_developing = 1 (developing)
*   Status_developing = 0 (developed)



In [ ]:
df = pd.get_dummies(df, columns=['Status'], drop_first=True, dtype=int)
#check the first 10 rows
print(df.head(10))

#check the data frame after one hot encoding
print(df.info())


   suicides/100k pop  Life expectancy  Adult Mortality  infant deaths  \
0               8.27             76.2               91              1   
1               5.38             76.2               91              1   
2               5.01             76.2               91              1   
3               3.94             76.2               91              1   
4               3.58             76.2               91              1   
5               3.27             76.2               91              1   
6               3.10             76.2               91              1   
7               3.04             76.2               91              1   
8               2.90             76.2               91              1   
9               2.28             76.2               91              1   

   Alcohol  percentage expenditure  Hepatitis B  Measles   BMI  \
0     5.28               41.822757           99       10  54.3   
1     5.28               41.822757           99       10  54.3  

# **Select Feature**
“How severe is the suicide rate in this country?”

In [ ]:
x = df.drop('suicides/100k pop', axis=1)
y = df['suicides/100k pop']


#A. **80-20 Train test split**

In [ ]:
from sklearn.model_selection import train_test_split

x_80_train, x_80_test, y_80_train, y_80_test = train_test_split(
    x, y, test_size=0.2, random_state=42
    )

print('Shape of X_train:', x_80_train.shape)
print('Shape of X_test:', x_80_test.shape)
print('Shape of y_train:', y_80_train.shape)
print('Shape of y_test:', y_80_test.shape)

Shape of X_train: (12211, 20)
Shape of X_test: (3053, 20)
Shape of y_train: (12211,)
Shape of y_test: (3053,)


# **Feature Scaling (80-20 split)**

In [ ]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
x_80_train_scaled = scaler.fit_transform(x_80_train)
x_80_test_scaled = scaler.transform(x_80_test)

print('Shape of scaled X_train:', x_80_train_scaled.shape)
print('Shape of scaled X_test:', x_80_test_scaled.shape)

Shape of scaled X_train: (12211, 20)
Shape of scaled X_test: (3053, 20)


# **Model Training (with 80-20 scaled data)**

In [ ]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.linear_model import LinearRegression
from sklearn.neighbors import KNeighborsRegressor
from sklearn.svm import SVR
from sklearn.neural_network import MLPRegressor
from sklearn.metrics import mean_squared_error, mean_absolute_error
import pandas as pd
import numpy as np

models = {
    'Random Forest': RandomForestRegressor(random_state=42),
    'Linear Regression': LinearRegression(),
    'K-Nearest Neighbors': KNeighborsRegressor(),
    'Support Vector Machine': SVR(),
    'Neural Network': MLPRegressor(max_iter=1000, random_state=42) # Increased max_iter for convergence
}

results_scaled_80_20 = []

for name, model in models.items():
    print(f"\nTraining {name}...")
    model.fit(x_80_train_scaled, y_80_train)
    y_pred = model.predict(x_80_test_scaled)

    #evaluate
    rmse = np.sqrt(mean_squared_error(y_80_test, y_pred))
    mae = mean_absolute_error(y_80_test, y_pred)

    results_scaled_80_20.append({
        'Model': name,
        'RMSE': rmse,
        'MAE': mae
    })

results_scaled_80_20_df = pd.DataFrame(results_scaled_80_20)
print("\nModel Performance (80-20 Split, Scaled Features):")
display(results_scaled_80_20_df)


Training Random Forest...

Training Linear Regression...

Training K-Nearest Neighbors...

Training Support Vector Machine...

Training Neural Network...

Model Performance (80-20 Split, Scaled Features):


Model       RMSE        MAE
0           Random Forest  17.384059  11.103800
1       Linear Regression  17.643497  11.582805
2     K-Nearest Neighbors  18.085607  11.307670
3  Support Vector Machine  18.611289  10.038649
4          Neural Network  16.493129  10.774773

#B. **70-30 Train test split**

In [ ]:

x_70_train, x_70_test, y_70_train, y_70_test = train_test_split(
    x, y, test_size=0.3, random_state=42
    )

print('Shape of X_train:', x_70_train.shape)
print('Shape of X_test:', x_70_test.shape)
print('Shape of y_train:', y_70_train.shape)
print('Shape of y_test:', y_70_test.shape)

Shape of X_train: (10684, 20)
Shape of X_test: (4580, 20)
Shape of y_train: (10684,)
Shape of y_test: (4580,)


# **Feature Scaling (70-30 split)**

In [ ]:
from sklearn.preprocessing import StandardScaler

scaler_70_30 = StandardScaler()
x_70_train_scaled = scaler_70_30.fit_transform(x_70_train)
x_70_test_scaled = scaler_70_30.transform(x_70_test)

print('Shape of scaled X_train (70-30 split):', x_70_train_scaled.shape)
print('Shape of scaled X_test (70-30 split):', x_70_test_scaled.shape)

Shape of scaled X_train (70-30 split): (10684, 20)
Shape of scaled X_test (70-30 split): (4580, 20)


# **Model Training (with 70-30 scaled data)**

In [ ]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.linear_model import LinearRegression
from sklearn.neighbors import KNeighborsRegressor
from sklearn.svm import SVR
from sklearn.neural_network import MLPRegressor
from sklearn.metrics import mean_squared_error, mean_absolute_error
import pandas as pd
import numpy as np

models_70_30 = {
    'Random Forest': RandomForestRegressor(random_state=42),
    'Linear Regression': LinearRegression(),
    'K-Nearest Neighbors': KNeighborsRegressor(),
    'Support Vector Machine': SVR(),
    'Neural Network': MLPRegressor(max_iter=1000, random_state=42) # Increased max_iter for convergence
}

results_scaled_70_30 = []

for name, model in models_70_30.items():
    print(f"\nTraining {name}...")
    model.fit(x_70_train_scaled, y_70_train)
    y_pred_70_30 = model.predict(x_70_test_scaled)

    #evaluate
    rmse_70_30 = np.sqrt(mean_squared_error(y_70_test, y_pred_70_30))
    mae_70_30 = mean_absolute_error(y_70_test, y_pred_70_30)

    results_scaled_70_30.append({
        'Model': name,
        'RMSE': rmse_70_30,
        'MAE': mae_70_30
    })

results_scaled_70_30_df = pd.DataFrame(results_scaled_70_30)
print("\nModel Performance (70-30 Split, Scaled Features):")
display(results_scaled_70_30_df)


Training Random Forest...

Training Linear Regression...

Training K-Nearest Neighbors...

Training Support Vector Machine...

Training Neural Network...

Model Performance (70-30 Split, Scaled Features):


Model       RMSE        MAE
0           Random Forest  17.218310  11.070457
1       Linear Regression  17.330860  11.478600
2     K-Nearest Neighbors  17.754772  11.252427
3  Support Vector Machine  18.236427   9.867941
4          Neural Network  16.231541  10.696761

# **Hyperparameter Tuning (80-20 Split)**

# **Hyperparameter Tuning with GridSearchCV (80-20 Split)**

In [ ]:
from sklearn.model_selection import GridSearchCV
from sklearn.ensemble import RandomForestRegressor

# Random Forest Regressor Grid Search
param_grid_rf = {
    'n_estimators': [50, 100, 150],
    'max_depth': [10, 20, None],
    'min_samples_leaf': [1, 2, 4]
}
rf_grid = GridSearchCV(RandomForestRegressor(random_state=42), param_grid_rf, cv=3, scoring='neg_mean_squared_error', n_jobs=-1, verbose=1)
rf_grid.fit(x_80_train_scaled, y_80_train)
best_rf_model_grid = rf_grid.best_estimator_
print("Best Random Forest (Grid Search) Parameters:", rf_grid.best_params_)

Fitting 3 folds for each of 27 candidates, totalling 81 fits
Best Random Forest (Grid Search) Parameters: {'max_depth': 10, 'min_samples_leaf': 4, 'n_estimators': 50}


In [ ]:
from sklearn.neural_network import MLPRegressor

# Neural Network Regressor Grid Search
param_grid_mlp = {
    'hidden_layer_sizes': [(50,), (100,), (50, 50)],
    'activation': ['relu', 'tanh'],
    'alpha': [0.0001, 0.001]
}
mlp_grid = GridSearchCV(MLPRegressor(max_iter=1000, random_state=42), param_grid_mlp, cv=3, scoring='neg_mean_squared_error', n_jobs=-1, verbose=1)
mlp_grid.fit(x_80_train_scaled, y_80_train)
best_mlp_model_grid = mlp_grid.best_estimator_
print("Best Neural Network (Grid Search) Parameters:", mlp_grid.best_params_)

Fitting 3 folds for each of 12 candidates, totalling 36 fits
Best Neural Network (Grid Search) Parameters: {'activation': 'relu', 'alpha': 0.0001, 'hidden_layer_sizes': (100,)}


# **Hyperparameter Tuning with RandomizedSearchCV (80-20 Split)**

In [ ]:
from sklearn.model_selection import RandomizedSearchCV
from scipy.stats import randint, uniform

# Random Forest Regressor Randomized Search
param_dist_rf = {
    'n_estimators': randint(50, 200),
    'max_depth': [10, 20, 30, None],
    'min_samples_leaf': randint(1, 5)
}
rf_random = RandomizedSearchCV(RandomForestRegressor(random_state=42), param_distributions=param_dist_rf, n_iter=10, cv=3, scoring='neg_mean_squared_error', n_jobs=-1, random_state=42, verbose=1)
rf_random.fit(x_80_train_scaled, y_80_train)
best_rf_model_random = rf_random.best_estimator_
print("Best Random Forest (Randomized Search) Parameters:", rf_random.best_params_)

Fitting 3 folds for each of 10 candidates, totalling 30 fits
Best Random Forest (Randomized Search) Parameters: {'max_depth': 10, 'min_samples_leaf': 4, 'n_estimators': 107}


In [ ]:
from sklearn.neural_network import MLPRegressor

# Neural Network Regressor Randomized Search
param_dist_mlp = {
    'hidden_layer_sizes': [(randint.rvs(50, 150),), (randint.rvs(50, 100), randint.rvs(20, 70))],
    'activation': ['relu', 'tanh'],
    'alpha': uniform(0.00001, 0.001)
}
mlp_random = RandomizedSearchCV(MLPRegressor(max_iter=1000, random_state=42), param_distributions=param_dist_mlp, n_iter=10, cv=3, scoring='neg_mean_squared_error', n_jobs=-1, random_state=42, verbose=1)
mlp_random.fit(x_80_train_scaled, y_80_train)
best_mlp_model_random = mlp_random.best_estimator_
print("Best Neural Network (Randomized Search) Parameters:", mlp_random.best_params_)

Fitting 3 folds for each of 10 candidates, totalling 30 fits
Best Neural Network (Randomized Search) Parameters: {'activation': 'relu', 'alpha': np.float64(0.0007896910002727694), 'hidden_layer_sizes': (65,)}


# **Evaluation**

In [ ]:
from sklearn.metrics import mean_squared_error, mean_absolute_error
import numpy as np
import pandas as pd

results = []

# Evaluate Best Random Forest Model (from GridSearchCV)
y_pred_rf_grid = best_rf_model_grid.predict(x_80_test_scaled)
rmse_rf_grid = np.sqrt(mean_squared_error(y_80_test, y_pred_rf_grid))
mae_rf_grid = mean_absolute_error(y_80_test, y_pred_rf_grid)
results.append({'Model': 'Random Forest (Grid)', 'RMSE': rmse_rf_grid, 'MAE': mae_rf_grid})

# Evaluate Best Neural Network Model (from GridSearchCV)
y_pred_mlp_grid = best_mlp_model_grid.predict(x_80_test_scaled)
rmse_mlp_grid = np.sqrt(mean_squared_error(y_80_test, y_pred_mlp_grid))
mae_mlp_grid = mean_absolute_error(y_80_test, y_pred_mlp_grid)
results.append({'Model': 'Neural Network (Grid)', 'RMSE': rmse_mlp_grid, 'MAE': mae_mlp_grid})

# Evaluate Best Random Forest Model (from Randomized Search)
y_pred_rf_random = best_rf_model_random.predict(x_80_test_scaled)
rmse_rf_random = np.sqrt(mean_squared_error(y_80_test, y_pred_rf_random))
mae_rf_random = mean_absolute_error(y_80_test, y_pred_rf_random)
results.append({'Model': 'Random Forest (Randomized)', 'RMSE': rmse_rf_random, 'MAE': mae_rf_random})

# Evaluate Best Neural Network Model (from Randomized Search)
y_pred_mlp_random = best_mlp_model_random.predict(x_80_test_scaled)
rmse_mlp_random = np.sqrt(mean_squared_error(y_80_test, y_pred_mlp_random))
mae_mlp_random = mean_absolute_error(y_80_test, y_pred_mlp_random)
results.append({'Model': 'Neural Network (Randomized)', 'RMSE': rmse_mlp_random, 'MAE': mae_mlp_random})

# Create a DataFrame to display results
tuned_results = pd.DataFrame(results)
print("\nPerformance of Tuned Models on 80-20 Test Set (Grid vs. Randomized Search):")
display(tuned_results)


Performance of Tuned Models on 80-20 Test Set (Grid vs. Randomized Search):


Model       RMSE        MAE
0         Random Forest (Grid)  16.946118  10.851623
1        Neural Network (Grid)  16.493129  10.774773
2   Random Forest (Randomized)  16.941014  10.845848
3  Neural Network (Randomized)  16.484933  10.774292

# **XAI EXECUTION**

In [ ]:
# =====================================
# XAI (SHAP) – Regression (ANN, Fused Dataset)
# =====================================

import shap
import numpy as np
import matplotlib.pyplot as plt

# -------------------------------------
# Select the final model to explain
# -------------------------------------
final_model = best_mlp_model_grid   # <- CONSISTENT with your evaluation

# Feature names from original dataframe
feature_names = x.columns.tolist()

# -------------------------------------
# Background data (small subset)
# -------------------------------------
np.random.seed(42)
background = x_80_test_scaled[
    np.random.choice(x_80_test_scaled.shape[0], 100, replace=False)
]

# -------------------------------------
# Create SHAP explainer (REGRESSION)
# -------------------------------------
explainer = shap.KernelExplainer(
    final_model.predict,
    background
)

# -------------------------------------
# Compute SHAP values
# -------------------------------------
X_explain = x_80_test_scaled[:50]
shap_values = explainer.shap_values(X_explain)

# -------------------------------------
# Global Feature Importance (Bar Plot)
# -------------------------------------
shap.summary_plot(
    shap_values,
    X_explain,
    feature_names=feature_names,
    plot_type="bar"
)

# -------------------------------------
# SHAP Summary Plot (Impact + Direction)
# -------------------------------------
shap.summary_plot(
    shap_values,
    X_explain,
    feature_names=feature_names
)

# -------------------------------------
# Local Explanation (Single Instance)
# -------------------------------------
sample_index = 0

shap.plots.waterfall(
    shap.Explanation(
        values=shap_values[sample_index],
        base_values=explainer.expected_value,
        data=X_explain[sample_index],
        feature_names=feature_names
    )
)


# **Using AutoGluon for AutoML Regression**

install AutoGluon.

In [ ]:
!pip install autogluon

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.4/42.4 kB 1.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.6/43.6 kB 3.0 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.6/43.6 kB 3.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
INFO: pip is looking at multiple versions of opentelemetry-sdk to determine which version is compatible with other requirements. This could take a while.
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 259.5/259.5 kB 12.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
INFO: pip is looking at multiple versions of openxlab to determine which version is compatible with other requirements. This could take a while.
INFO: pip is still looking at multiple versions of openxlab to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter co

Now that AutoGluon is installed, we can use it to perform AutoML regression. We will use the `TabularPredictor` for this task, which automatically handles preprocessing, model selection, and hyperparameter tuning.

We'll use an 80-20 train-test split for this, similar to your previous manual modeling approach. AutoGluon's `fit` method will handle the training and internal validation.

In [ ]:
from autogluon.tabular import TabularDataset, TabularPredictor
from sklearn.model_selection import train_test_split
import pandas as pd

# Ensure df_pycaret is defined, if not, reconstruct it from x and y
if 'df_pycaret' not in locals():
    df_pycaret = pd.concat([x, y], axis=1)

# Split the data into training and test sets (80-20 split)
train_data, test_data = train_test_split(df_pycaret, test_size=0.2, random_state=42)

# Initialize TabularPredictor
predictor = TabularPredictor(label='suicides/100k pop', eval_metric='rmse', path='AutogluonModels').fit(train_data, presets='best_quality')

# Make predictions on the test data
y_pred_autogluon = predictor.predict(test_data)

# Evaluate the model
from sklearn.metrics import mean_squared_error, mean_absolute_error
import numpy as np

rmse_autogluon = np.sqrt(mean_squared_error(test_data['suicides/100k pop'], y_pred_autogluon))
mae_autogluon = mean_absolute_error(test_data['suicides/100k pop'], y_pred_autogluon)

print(f"\nAutoGluon Model Performance (80-20 Split):")
print(f"RMSE: {rmse_autogluon:.4f}")
print(f"MAE: {mae_autogluon:.4f}")

Verbosity: 2 (Standard Logging)
=================== System Info ===================
AutoGluon Version:  1.5.0
Python Version:     3.12.12
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #1 SMP Thu Oct  2 10:42:05 UTC 2025
CPU Count:          2
Pytorch Version:    2.9.0+cpu
CUDA Version:       CUDA is not available
Memory Avail:       10.31 GB / 12.67 GB (81.4%)
Disk Space Avail:   85.34 GB / 107.72 GB (79.2%)
Presets specified: ['best_quality']
Using hyperparameters preset: hyperparameters='zeroshot'
Setting dynamic_stacking from 'auto' to True. Reason: Enable dynamic_stacking when use_bag_holdout is disabled. (use_bag_holdout=False)
Stack configuration (auto_stack=True): num_stack_levels=1, num_bag_folds=8, num_bag_sets=1
DyStack is enabled (dynamic_stacking=True). AutoGluon will try to determine whether the input data is affected by stacked overfitting and enable or disable stacking as a consequence.
	This is used to identify the optimal `num_stack_levels` va


AutoGluon Model Performance (80-20 Split):
RMSE: 16.2986
MAE: 10.5098


## Summary:

### Q&A
The most important features identified by the SHAP analysis for the best-performing Neural Network models, and their influence on the prediction of `suicides/100k pop`, are:
*   **`gdp_per_capita ($)`**: Consistently showed a strong positive correlation, meaning higher GDP per capita values tend to increase the predicted suicide rate.
*   **`Life expectancy`**: Generally associated with lower predicted `suicides/100k pop`; higher life expectancy leads to lower suicide rates.
*   **`Income composition of resources`**: Similar to `gdp_per_capita ($)`, higher values for this feature tended to increase the predicted `suicides/100k pop`.
*   **`Schooling`**: Higher schooling years were generally associated with lower predicted `suicides/100k pop`.
*   **`Adult Mortality`**: Higher adult mortality values seemed to increase the predicted `suicides/100k pop`.

### Data Analysis Key Findings
*   The **Neural Network (Randomized)** model was identified as the best-performing model with the lowest Root Mean Squared Error (RMSE) of 16.484933 and a Mean Absolute Error (MAE) of 10.774292.
*   SHAP analysis for both Neural Network models (Grid Search and Randomized Search) consistently highlighted **`gdp_per_capita ($)`** as a strong positive predictor, indicating higher GDP per capita is associated with increased suicide rates.
*   **`Life expectancy`** and **`Schooling`** were identified as having a negative correlation with `suicides/100k pop`, suggesting that higher values in these features lead to lower predicted suicide rates.
*   **`Income composition of resources`** and **`Adult Mortality`** also emerged as significant features, with higher values typically corresponding to increased predicted suicide rates.
*   Both Neural Network models showed strong agreement on the ranking and directional impact of these key features.

### Insights or Next Steps
*   **Socio-economic factors are complex drivers**: The findings suggest that economic indicators like `gdp_per_capita ($)` and `Income composition of resources` are highly influential, with a seemingly counter-intuitive positive correlation with suicide rates. This warrants further investigation into the nuances of prosperity and societal well-being.
*   **Health and education as protective factors**: Features such as `Life expectancy` and `Schooling` consistently show an inverse relationship with suicide rates, underscoring the protective role of robust health and education systems in mitigating suicide risk.
